### PR TITLE
Hotfix for password checking fail

### PR DESCRIPTION
**Pull request by dev4u**
可能存在`iPSK:uPSK`格式的密码。
如果要在客户端检测，要先以`:`做分组，然后对分组后的psk做检测。
Ref: https://github.com/shadowsocks/shadowsocks-android/commit/243cafd41d0712da1f5693067f1f9b0fc6408645#r102304121

### CODE REVIEW

**Comment by Mygod** [comment]
Wow imagining not putting all of this crap in a plugin. 😒

File: core/src/main/java/com/github/shadowsocks/bg/ProxyInstance.kt [hunk 1]
```diff
@@ -50,13 +51,15 @@ class ProxyInstance(val profile: Profile, private val route: String = profile.ro
...
         }
         // check the key format for aead-2022-cipher
-        require(profile.method !in setOf(
-            "2022-blake3-aes-128-gcm",
-            "2022-blake3-aes-256-gcm",
-            "2022-blake3-chacha20-poly1305",
-        ) || Base64.decode(profile.password, Base64.DEFAULT).size in arrayOf(16, 32)) {
-            "The Base64 Key is invalid."
-        }
+        profile.password.split(":").forEach(Consumer {
```
Comment (Mygod) [inline core/src/main/java/com/github/shadowsocks/bg/ProxyInstance.kt:54]: I don't think you need to use `Consumer` in Kotlin?

**Comment by dev4u** [comment]
> Wow imagining not putting all of this crap in a plugin. 😒

😄这是病，已安排治疗。希望能控制下次不病发。

**Comment by dev4u** [comment]
@Mygod 这个应该可以审查了 ヽ(^。^)丿

**Comment by suleymanisangulyyew43-oss** [comment]
Shadowsock

**Pull request merged**